hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mob/mapreduce/MobRefReporter.java

```diff
@@ -0,0 +1,501 @@
+ *         all of our rows have somewhere between 1 - 9 cells.
+ *   * PROBLEM_ROWS_PER_FILE_DIGITS: - this counter group gives a histogram of the order of
+ *         magnitude of the number of rows in each of the hfiles with a problem. e.g. in the
+ *         example there are 2 hfiles and they each have the same order of magnitude number of rows,
+ *         namely 3 digits worth i.e. 100-999.
+ *   * SIZE_PER_CELL_DIGITS: - this counter group gives a histogram of the order of magnitude of
+ *         the size of mob values according to our reference cells. e.g. in the example above we
+ *         have between 5 and 7 digits in our cell sizes i.e. 10,000 - 9,999,999 bytes. From this
+ *         histogram we can also see that _most_ cells are 100,000 - 199,000 bytes and the smaller
+ *         and bigger ones are outliers making up less than 2% of mob cells.
+ *   * SIZE_PER_ROW_DIGITS: - this counter group gives a histogram of the order of magnitude of the
+ *         size of
+ *
+ * Generates a report that gives one file status per line, with tabs dividing fields.
+ *
+ * <pre>
+ * RESULT OF LOOKUP	FILE REF	comma seperated, base64 encoded rows when there's a problem
+ * </pre>
+ *
+ * e.g.
+ *
+ * <pre>
+ * MOB DIR	09c576e28a65ed2ead0004d192ffaa382019110184b30a1c7e034573bf8580aef8393402
+ * MISSING FILE    28e252d7f013973174750d483d358fa020191101f73536e7133f4cd3ab1065edf588d509        MmJiMjMyYzBiMTNjNzc0OTY1ZWY4NTU4ZjBmYmQ2MTUtNTIz,MmEzOGE0YTkzMTZjNDllNWE4MzM1MTdjNDVkMzEwNzAtODg=
+ * </pre>
+ *
+ * Possible results are listed; the first three indicate things are working properly.
+ *   * MOB DIR - the reference is in the normal MOB area for the given table and CF
+ *   * HLINK TO ARCHIVE FOR SAME TABLE - the reference is present in the archive area for this
+ *         table and CF
+ *   * HLINK TO ARCHIVE FOR OTHER TABLE - the reference is present in a different table and CF,
+ *         either in the MOB or archive areas (e.g. from a snapshot restore or clone)
+ *   * ARCHIVE WITH HLINK BUT NOT FROM OUR TABLE - the reference is currently present in the archive
+ *         area for this table and CF, but it is kept there because a _different_ table has a
+ *         reference to it (e.g. from a snapshot clone). If these other tables are removed then
+ *         the file will likely be deleted unless there is a snapshot also referencing it.
+ *   * ARCHIVE BUT NO HLINKS - the reference is currently present in the archive for this table and
+ *         CF, but there are no references present to prevent its removal. Unless it is newer than
+ *         the general TTL (default 5 minutes) or referenced in a snapshot it will be subject to
+ *         cleaning.
+ *   * ARCHIVE BUT IOE WHILE CHECKING HLINKS - Check the job logs to see why things failed while
+ *         looking for why this file is being kept around.
+ *   * MISSING FILE - We couldn't find the reference on the FileSystem. Either there is dataloss due
```

Comment:
I appreciate all the info I need is here but it is really confusing; understandably the system should not be in this state too. Perhaps, leave this with a shorter statement such as, HBase should not have this happen, please file a ticket with the HBase project if you find yourself here? Or more usable, could you provide pointer to a description of how MOB works in the reference guide (containing how the server-side tag, HFileLink's and HFiles go together) and suggest the admin should walk through the states a MOB HFileLink should be in and determine where the missing piece is to manually fix it?

Reply:
Maybe the "get help" message is better placed in that message in the job's log?

Reply:
Are you suggesting new output names, new Java variable names, or new prefixes to the same?

We need the level of granularity already present to diagnose what's going on with a running system. So reducing the set of outputs would hinder that. I could maybe add a `--verbose` or `--simplified` flag to give the option to pick between "tell me if something's wrong" vs "tell me what is wrong". Maybe?

Reply:
The granularity between the various possible results is great and all the messages should be kept in. Only the description for MISSING_FILE seems like too much and I would love a more thorough explanation of MOB to refer to as an admin. And architectural description would be better placed somewhere else (e.g. the reference guide or somewhere more obviously central to MOB design).

For MISSING_FILE here, simply saying MOB's broken and suggesting what components are missing/misaligned and pointing someone to the more thorough architecture reference for them to debug more deeply would be my preferred approach, if I were the admin hitting this.

Reply:
I think I can do that. let me make a pass at updated wording.